src/main/java/backend/Logic.java

```diff
@@ -231,11 +233,26 @@ public void removeUnusedModels(Set<String> reposInUse) {
                 .forEach(models::removeRepoModelById);
     }
 
+    /**
+     * Updates events related to remaining rate.
+     * @param rateLimits The api rate limits for updating of the remaining rate.
+     * @return Returns the rateLimits instance
+     */
     public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
         uiManager.updateRateLimits(rateLimits);
         return rateLimits;
     }
 
+    /**
+     * Updates the sync refresh rate of updating on the current data store.
+     * @param rateLimits The api rate limits for calculation of the refresh rate.
+     * @return Returns the rateLimits instance
+     */
+    public ImmutablePair<Integer, Long> updateSyncRefreshRate(ImmutablePair<Integer, Long> rateLimits) {
+        uiManager.updateSyncRefreshRate(rateLimits);
+        return rateLimits;
+    }
+
```

Comment:
I have revert back to the old commit. Two events is necessary as one event is for updating of ui component and one is updating of sync refresh rate. 

Updating of ui component can occurs at any time based on user usage. But updating of sync refresh rate should be only called once in every trigger of the refresh timer.    

I have changed the header comments for the function. It should be more clear now.